agent/agentssh/agentssh.go

```diff
@@ -1099,3 +1100,32 @@ func userHomeDir() (string, error) {
 	}
 	return u.HomeDir, nil
 }
+
+// UpdateHostSigner updates the host signer with a new key generated from the provided seed.
+// This function replaces the previously used host key with the newly generated one.
```

Comment:
It could be useful to mention here that gliderlabs/ssh matches on key type and only allows one/type. At least for me I was wondering how this works and had to dive into the docs to find:

```
If an existing host key exists with the same algorithm, it is overwritten
```

Reply:
Yeah, the wording from `gliderlabs/ssh` is clearer.